# Setup

In [ ]:
# Docker and Milvus Standalone

# 1. wget https://github.com/milvus-io/milvus/releases/download/v2.0.2/milvus-standalone-docker-compose.yml -O docker-compose.yml
# 2. https://docs.docker.com/compose/install/
# 3. Start: sudo docker-compose up -d 
#     Check:sudo docker-compose ps
#     Stop: sudo docker-compose down
#     Delete data after stop: sudo rm -rf  volumes


In [2]:
# Python 3.7.1 or later is required

In [ ]:
# pymilvus

# python3 -m pip install pymilvus==2.0.2
#  Verify: python3 -c "from pymilvus import Collection"

# Import Libraries

In [6]:
from pymilvus import (
    connections,
    utility,
    FieldSchema,
    CollectionSchema,
    DataType,
    Collection,
)
import time
import numpy as np

# Read Embeddings

### This notebook was run on EC2 (Deep Learning AMI) to connect to Milvus servers for the purpose of vector search. Input to this notebook can be found here: https://drive.google.com/drive/folders/1M4_SrEPMRo9PSNLWzPHPHN0MpPD4uzjB?usp=sharing

In [3]:
import os

In [7]:
X_train_features = np.loadtxt(os.path.join("0004",'X_train_features.txt'))
X_test_features = np.loadtxt(os.path.join("0004",'X_test_features.txt'))

In [8]:
num_entities, dim = X_train_features.shape

In [9]:
y_train = np.loadtxt(os.path.join("0004",'y_train.txt'))
y_test = np.loadtxt(os.path.join("0004",'y_test.txt'))

In [10]:
train_images = np.loadtxt(os.path.join("0004",'train_images.txt'))
test_images = np.loadtxt(os.path.join("0004",'test_images.txt'))

In [11]:
X_train_features.shape

(10586, 1792)

In [12]:
y_train.astype(int)

array([5543,  896,  886, ..., 2310,  802,  385])

# Connect to Server

In [11]:
connections.connect("default", host="localhost", port="19530")

# Create Collection

In [13]:
has = utility.has_collection("facenet")
print(f"Does collection facenet exist in Milvus: {has}")

Does collection facenet exist in Milvus: False


In [17]:
fmt = "\n=== {:30} ===\n"

In [18]:
fields = [
    FieldSchema(name="pk", dtype=DataType.INT64,
                is_primary=True, auto_id=False, description="primary id"),
    FieldSchema(name="label", dtype=DataType.INT64, description="label"),
    FieldSchema(name="image", dtype=DataType.INT64, description="image_number"),
    FieldSchema(name="embeddings", dtype=DataType.FLOAT_VECTOR,
                dim=dim, description="vector")
]

schema = CollectionSchema(fields, "Embeddings for facenet")
print(fmt.format("Create collection `embed`"))
embed = Collection("embed", schema,
                          using='default', shards_num=2)


=== Create collection `embed`      ===



In [24]:
print(fmt.format("Start inserting entities"))

entities = [
    [i for i in range(0, num_entities)],  # field pk
    y_train.astype(int), # labels
    train_images.astype(int), # images
    X_train_features,  # field embeddings
]
insert_result = embed.insert(entities)


=== Start inserting entities       ===



In [33]:
print(entities[3])

IOPub data rate exceeded.
The notebook server will temporarily stop sending output
to the client in order to avoid crashing it.
To change this limit, set the config variable
`--NotebookApp.iopub_data_rate_limit`.

Current values:
NotebookApp.iopub_data_rate_limit=1000000.0 (bytes/sec)
NotebookApp.rate_limit_window=3.0 (secs)



In [25]:
print(f"Number of Entities: {embed.num_entities}")

Number of Entities: 10586


# Build Index on Entities

In [37]:
import math

print(fmt.format("Start Creating index IVF_FLAT"))

# optimal value of nlist is 4 * sqrt(num of enteries)
index = {
    "index_type": "IVF_FLAT",
    "metric_type": "L2",
    "params": {"nlist": int(4 * math.sqrt(num_entities))},
}
embed.create_index("embeddings", index)


=== Start Creating index IVF_FLAT  ===



Status(code=0, message='')

# Load to Memory

In [39]:
print(fmt.format("Start loading"))

embed.load()


=== Start loading                  ===



# Vector Similarity Search

## Similartiy Search

In [40]:
# nprobe / nlist = 1/4
search_params = {
    "metric_type": "l2",
    "params": {"nprobe": 330},
}

In [87]:
start_time = time.time()
# here limit is set to 7 to get 7 similar images later will be chaged to 20
result = embed.search(X_test_features, "embeddings", search_params, limit=7, output_fields=["label", "image", "pk"])
end_time = time.time()

In [88]:
print(f'Time Taken: {end_time - start_time}s')

Time Taken: 5.631507158279419s


# Result 

In [89]:
final_result = dict()

# final_result = {
#     'label_imgNumber': {
#         "label_imgNumber": distance,
#         "label_imgNumber": distance,
#         "label_imgNumber": distance,
        
#     },
#     ...
# }

In [90]:
y_test

array([ 173., 4836., 1428., ..., 5354., 3109., 1393.])

In [91]:
test_images

array([ 2., 26.,  2., ...,  1.,  3., 41.])

In [92]:
def get_query_number(number):
    temp = str(number)
    while len(temp) != 4:
        temp = '0' + temp
    return temp

In [93]:
index = 0
for hits in result:
    print(f'Query Number: {index}')
    
    query_label = int(y_test[index])
    query_imgNumber = int(test_images[index])
    
    current_query_id = str(query_label) + '_' + get_query_number(query_imgNumber)
    
    final_result[current_query_id] = dict()
    
    for hit in hits:
        result_label = int(hit.entity.get('label'))
        result_imgNumber = int(hit.entity.get('image'))
        
        current_result_id = str(result_label) + '_' + get_query_number(result_imgNumber)
        
        final_result[current_query_id][current_result_id] = hit.distance
        
        print(f"hit: {hit}, Label: {hit.entity.get('label')}, Image: {hit.entity.get('image')}")
    
    index += 1

Query Number: 0
hit: (distance: 26.057170867919922, id: 119), Label: 173, Image: 1
hit: (distance: 127.43841552734375, id: 1957), Label: 999, Image: 1
hit: (distance: 130.5911865234375, id: 7793), Label: 1105, Image: 2
hit: (distance: 130.891357421875, id: 8090), Label: 2368, Image: 1
hit: (distance: 131.17236328125, id: 4305), Label: 1610, Image: 1
hit: (distance: 135.7631378173828, id: 3508), Label: 3250, Image: 1
hit: (distance: 136.7291259765625, id: 3623), Label: 2368, Image: 2
Query Number: 1
hit: (distance: 37.43260192871094, id: 4727), Label: 4836, Image: 8
hit: (distance: 53.6215934753418, id: 6419), Label: 4836, Image: 13
hit: (distance: 59.88230895996094, id: 3374), Label: 4836, Image: 19
hit: (distance: 60.98465347290039, id: 10396), Label: 4836, Image: 5
hit: (distance: 63.946720123291016, id: 6971), Label: 4836, Image: 9
hit: (distance: 66.75064086914062, id: 3127), Label: 4836, Image: 23
hit: (distance: 67.14411926269531, id: 5197), Label: 4836, Image: 17
Query Number: 2

hit: (distance: 233.34690856933594, id: 6413), Label: 3467, Image: 5
Query Number: 126
hit: (distance: 87.2083511352539, id: 6170), Label: 2075, Image: 4
hit: (distance: 89.87103271484375, id: 5747), Label: 2075, Image: 3
hit: (distance: 97.59793853759766, id: 8776), Label: 2075, Image: 2
hit: (distance: 152.41738891601562, id: 8079), Label: 2878, Image: 1
hit: (distance: 163.1081085205078, id: 7988), Label: 3212, Image: 1
hit: (distance: 163.1358642578125, id: 9915), Label: 4062, Image: 1
hit: (distance: 166.01821899414062, id: 7886), Label: 3335, Image: 3
Query Number: 127
hit: (distance: 65.844482421875, id: 3267), Label: 5458, Image: 131
hit: (distance: 77.38188171386719, id: 5088), Label: 5458, Image: 38
hit: (distance: 78.87896728515625, id: 7206), Label: 5458, Image: 25
hit: (distance: 79.29905700683594, id: 6590), Label: 5458, Image: 4
hit: (distance: 83.17660522460938, id: 9302), Label: 5458, Image: 40
hit: (distance: 84.04142761230469, id: 4982), Label: 5458, Image: 33
hit: (

hit: (distance: 150.10186767578125, id: 8414), Label: 444, Image: 1
hit: (distance: 158.92657470703125, id: 2094), Label: 584, Image: 1
hit: (distance: 161.87489318847656, id: 998), Label: 2901, Image: 1
hit: (distance: 164.2852020263672, id: 2253), Label: 444, Image: 2
hit: (distance: 167.12033081054688, id: 832), Label: 4898, Image: 1
hit: (distance: 173.80010986328125, id: 5427), Label: 3411, Image: 3
Query Number: 156
hit: (distance: 66.10835266113281, id: 584), Label: 4114, Image: 5
hit: (distance: 90.63130950927734, id: 6401), Label: 4114, Image: 2
hit: (distance: 92.84676361083984, id: 4252), Label: 4114, Image: 8
hit: (distance: 96.28128051757812, id: 8828), Label: 4114, Image: 13
hit: (distance: 99.16423797607422, id: 1248), Label: 4114, Image: 11
hit: (distance: 102.51397705078125, id: 8212), Label: 4114, Image: 7
hit: (distance: 114.55485534667969, id: 6059), Label: 4114, Image: 12
Query Number: 157
hit: (distance: 105.675537109375, id: 617), Label: 2500, Image: 18
hit: (dis

Query Number: 284
hit: (distance: 38.23125076293945, id: 5902), Label: 417, Image: 3
hit: (distance: 38.57796859741211, id: 962), Label: 417, Image: 12
hit: (distance: 49.595787048339844, id: 6594), Label: 417, Image: 5
hit: (distance: 53.27894592285156, id: 8172), Label: 417, Image: 2
hit: (distance: 56.886817932128906, id: 9269), Label: 417, Image: 22
hit: (distance: 59.8803825378418, id: 7366), Label: 417, Image: 1
hit: (distance: 74.66339111328125, id: 8577), Label: 417, Image: 7
Query Number: 285
hit: (distance: 125.2592544555664, id: 8409), Label: 4859, Image: 1
hit: (distance: 134.16921997070312, id: 4040), Label: 1091, Image: 2
hit: (distance: 135.94174194335938, id: 6743), Label: 4672, Image: 2
hit: (distance: 139.41748046875, id: 651), Label: 3995, Image: 5
hit: (distance: 143.574462890625, id: 4722), Label: 3995, Image: 3
hit: (distance: 145.25311279296875, id: 10193), Label: 2891, Image: 1
hit: (distance: 147.46571350097656, id: 4960), Label: 779, Image: 1
Query Number: 286

hit: (distance: 131.66421508789062, id: 6514), Label: 3318, Image: 1
hit: (distance: 134.99391174316406, id: 4409), Label: 423, Image: 1
hit: (distance: 157.23651123046875, id: 3567), Label: 1338, Image: 2
hit: (distance: 159.94357299804688, id: 6569), Label: 229, Image: 1
hit: (distance: 173.4142303466797, id: 3233), Label: 2637, Image: 1
hit: (distance: 174.1541748046875, id: 7460), Label: 1114, Image: 1
hit: (distance: 175.2963409423828, id: 4655), Label: 5264, Image: 1
Query Number: 314
hit: (distance: 116.52391052246094, id: 4225), Label: 4182, Image: 1
hit: (distance: 117.1749267578125, id: 4544), Label: 5694, Image: 1
hit: (distance: 117.57991027832031, id: 2128), Label: 3104, Image: 1
hit: (distance: 124.37317657470703, id: 642), Label: 5700, Image: 1
hit: (distance: 125.62275695800781, id: 7427), Label: 2186, Image: 4
hit: (distance: 125.86576843261719, id: 7974), Label: 4566, Image: 1
hit: (distance: 127.35232543945312, id: 1188), Label: 5608, Image: 2
Query Number: 315
hit: 

hit: (distance: 155.66867065429688, id: 5603), Label: 481, Image: 1
hit: (distance: 164.471435546875, id: 4292), Label: 1267, Image: 1
hit: (distance: 165.92626953125, id: 4783), Label: 5219, Image: 8
hit: (distance: 167.2408447265625, id: 8787), Label: 481, Image: 3
Query Number: 441
hit: (distance: 58.86305236816406, id: 3677), Label: 781, Image: 15
hit: (distance: 65.43730163574219, id: 9227), Label: 781, Image: 13
hit: (distance: 68.20507049560547, id: 4986), Label: 781, Image: 9
hit: (distance: 74.53215789794922, id: 9914), Label: 781, Image: 16
hit: (distance: 77.96800231933594, id: 2750), Label: 781, Image: 12
hit: (distance: 78.25784301757812, id: 7024), Label: 781, Image: 2
hit: (distance: 81.99131774902344, id: 4399), Label: 781, Image: 5
Query Number: 442
hit: (distance: 47.77399444580078, id: 4649), Label: 3201, Image: 21
hit: (distance: 47.975379943847656, id: 8981), Label: 3201, Image: 17
hit: (distance: 56.939414978027344, id: 1192), Label: 3201, Image: 6
hit: (distance:

hit: (distance: 166.71185302734375, id: 8764), Label: 4963, Image: 3
Query Number: 472
hit: (distance: 61.88454055786133, id: 6754), Label: 1964, Image: 1
hit: (distance: 65.06680297851562, id: 4501), Label: 1964, Image: 21
hit: (distance: 65.09230041503906, id: 4806), Label: 1964, Image: 2
hit: (distance: 65.68081665039062, id: 522), Label: 1964, Image: 16
hit: (distance: 69.26140594482422, id: 4605), Label: 1964, Image: 3
hit: (distance: 70.85484313964844, id: 7173), Label: 1964, Image: 15
hit: (distance: 73.13334655761719, id: 10497), Label: 1964, Image: 14
Query Number: 473
hit: (distance: 92.06900024414062, id: 8228), Label: 3702, Image: 22
hit: (distance: 99.85137939453125, id: 166), Label: 5704, Image: 10
hit: (distance: 100.93528747558594, id: 6549), Label: 5704, Image: 9
hit: (distance: 106.83505249023438, id: 1340), Label: 5250, Image: 4
hit: (distance: 108.39686584472656, id: 884), Label: 5704, Image: 11
hit: (distance: 108.74553680419922, id: 7836), Label: 5704, Image: 14
h

hit: (distance: 54.705604553222656, id: 5582), Label: 1393, Image: 9
hit: (distance: 66.00856018066406, id: 4674), Label: 1393, Image: 112
hit: (distance: 68.61687469482422, id: 3975), Label: 1393, Image: 37
hit: (distance: 76.46735382080078, id: 3688), Label: 1393, Image: 8
hit: (distance: 81.11050415039062, id: 8194), Label: 1393, Image: 51
hit: (distance: 81.56208801269531, id: 2195), Label: 1393, Image: 85
hit: (distance: 81.9244613647461, id: 7752), Label: 1393, Image: 32
Query Number: 601
hit: (distance: 123.56639862060547, id: 6799), Label: 3076, Image: 1
hit: (distance: 151.31484985351562, id: 8235), Label: 2284, Image: 1
hit: (distance: 153.51107788085938, id: 500), Label: 3919, Image: 2
hit: (distance: 159.44711303710938, id: 5609), Label: 3076, Image: 2
hit: (distance: 160.46929931640625, id: 9445), Label: 3796, Image: 1
hit: (distance: 160.96327209472656, id: 9635), Label: 2210, Image: 12
hit: (distance: 165.12063598632812, id: 10105), Label: 2489, Image: 1
Query Number: 60

hit: (distance: 167.34481811523438, id: 8780), Label: 2730, Image: 13
hit: (distance: 168.418701171875, id: 3819), Label: 2730, Image: 1
hit: (distance: 169.3143768310547, id: 10440), Label: 2730, Image: 6
hit: (distance: 169.42852783203125, id: 5962), Label: 2730, Image: 5
Query Number: 633
hit: (distance: 103.72005462646484, id: 3228), Label: 706, Image: 5
hit: (distance: 126.86265563964844, id: 3636), Label: 706, Image: 1
hit: (distance: 133.64199829101562, id: 9587), Label: 706, Image: 4
hit: (distance: 135.23410034179688, id: 771), Label: 706, Image: 6
hit: (distance: 141.60513305664062, id: 4010), Label: 3916, Image: 1
hit: (distance: 150.498291015625, id: 3107), Label: 706, Image: 11
hit: (distance: 150.7110595703125, id: 4221), Label: 706, Image: 12
Query Number: 634
hit: (distance: 100.62866973876953, id: 5241), Label: 1565, Image: 5
hit: (distance: 109.25238037109375, id: 3068), Label: 1565, Image: 4
hit: (distance: 176.23348999023438, id: 10490), Label: 2896, Image: 1
hit: (

hit: (distance: 151.453369140625, id: 1022), Label: 371, Image: 1
hit: (distance: 157.8193359375, id: 1621), Label: 4282, Image: 1
hit: (distance: 160.41444396972656, id: 1129), Label: 4825, Image: 2
hit: (distance: 161.2498779296875, id: 661), Label: 4825, Image: 4
hit: (distance: 161.40049743652344, id: 10427), Label: 92, Image: 1
hit: (distance: 163.42019653320312, id: 1910), Label: 277, Image: 1
Query Number: 762
hit: (distance: 84.7845458984375, id: 1619), Label: 5301, Image: 3
hit: (distance: 135.77720642089844, id: 6416), Label: 3737, Image: 11
hit: (distance: 138.313720703125, id: 129), Label: 3737, Image: 1
hit: (distance: 144.04119873046875, id: 8229), Label: 2469, Image: 14
hit: (distance: 148.44281005859375, id: 8467), Label: 4515, Image: 1
hit: (distance: 149.15830993652344, id: 8189), Label: 2469, Image: 16
hit: (distance: 151.58847045898438, id: 10294), Label: 1261, Image: 1
Query Number: 763
hit: (distance: 139.68701171875, id: 10341), Label: 4465, Image: 1
hit: (distan

hit: (distance: 182.84194946289062, id: 8258), Label: 1909, Image: 1
hit: (distance: 185.41065979003906, id: 10098), Label: 704, Image: 1
hit: (distance: 186.5255889892578, id: 5906), Label: 3335, Image: 2
Query Number: 800
hit: (distance: 90.69181060791016, id: 9721), Label: 3201, Image: 29
hit: (distance: 123.30469512939453, id: 8905), Label: 3201, Image: 24
hit: (distance: 125.59074401855469, id: 6070), Label: 3201, Image: 28
hit: (distance: 131.53125, id: 5722), Label: 3201, Image: 2
hit: (distance: 131.71658325195312, id: 3532), Label: 3201, Image: 5
hit: (distance: 140.14138793945312, id: 4649), Label: 3201, Image: 21
hit: (distance: 143.29493713378906, id: 8795), Label: 3201, Image: 14
Query Number: 801
hit: (distance: 168.35284423828125, id: 1366), Label: 1146, Image: 1
hit: (distance: 174.90460205078125, id: 9291), Label: 600, Image: 1
hit: (distance: 175.89642333984375, id: 6739), Label: 4776, Image: 2
hit: (distance: 178.50521850585938, id: 665), Label: 3831, Image: 3
hit: (

hit: (distance: 156.15234375, id: 8957), Label: 3202, Image: 1
hit: (distance: 163.42147827148438, id: 2063), Label: 302, Image: 6
hit: (distance: 166.4627685546875, id: 2892), Label: 4200, Image: 1
hit: (distance: 167.4069366455078, id: 1017), Label: 302, Image: 14
hit: (distance: 169.93641662597656, id: 7579), Label: 302, Image: 9
hit: (distance: 170.90081787109375, id: 8836), Label: 703, Image: 1
hit: (distance: 171.45603942871094, id: 2223), Label: 3411, Image: 1
Query Number: 925
hit: (distance: 70.3094711303711, id: 3748), Label: 5514, Image: 6
hit: (distance: 77.18331909179688, id: 2699), Label: 5514, Image: 3
hit: (distance: 91.04008483886719, id: 8979), Label: 5514, Image: 4
hit: (distance: 96.77951049804688, id: 2431), Label: 5514, Image: 7
hit: (distance: 106.79438781738281, id: 3429), Label: 5514, Image: 8
hit: (distance: 108.01000213623047, id: 4942), Label: 5514, Image: 2
hit: (distance: 138.73204040527344, id: 3861), Label: 1869, Image: 2
Query Number: 926
hit: (distance

hit: (distance: 121.62400817871094, id: 8011), Label: 3226, Image: 1
hit: (distance: 133.24795532226562, id: 8081), Label: 1046, Image: 1
hit: (distance: 138.24069213867188, id: 4672), Label: 2470, Image: 55
hit: (distance: 138.29844665527344, id: 5341), Label: 1308, Image: 1
hit: (distance: 140.08184814453125, id: 10180), Label: 4367, Image: 1
hit: (distance: 148.08071899414062, id: 6161), Label: 4606, Image: 1
hit: (distance: 149.26783752441406, id: 2550), Label: 2812, Image: 3
Query Number: 957
hit: (distance: 159.5947265625, id: 3423), Label: 1632, Image: 1
hit: (distance: 181.40550231933594, id: 5296), Label: 3567, Image: 4
hit: (distance: 184.26785278320312, id: 7297), Label: 1695, Image: 9
hit: (distance: 192.56005859375, id: 625), Label: 553, Image: 1
hit: (distance: 193.9080047607422, id: 2924), Label: 3625, Image: 3
hit: (distance: 194.04689025878906, id: 6488), Label: 2364, Image: 1
hit: (distance: 196.054443359375, id: 5946), Label: 2302, Image: 9
Query Number: 958
hit: (di

hit: (distance: 107.1962890625, id: 4562), Label: 1382, Image: 12
hit: (distance: 107.44561767578125, id: 5134), Label: 1382, Image: 11
hit: (distance: 128.11126708984375, id: 1626), Label: 896, Image: 2
Query Number: 1089
hit: (distance: 49.631526947021484, id: 9451), Label: 4657, Image: 1
hit: (distance: 166.07379150390625, id: 7059), Label: 5601, Image: 4
hit: (distance: 171.90159606933594, id: 8353), Label: 2173, Image: 18
hit: (distance: 173.2256317138672, id: 9768), Label: 5601, Image: 3
hit: (distance: 176.35980224609375, id: 2930), Label: 5601, Image: 2
hit: (distance: 176.5885009765625, id: 9655), Label: 578, Image: 1
hit: (distance: 177.16822814941406, id: 10368), Label: 2173, Image: 44
Query Number: 1090
hit: (distance: 61.566864013671875, id: 3409), Label: 4960, Image: 45
hit: (distance: 71.16447448730469, id: 4742), Label: 4960, Image: 4
hit: (distance: 73.24542236328125, id: 477), Label: 4960, Image: 36
hit: (distance: 74.73573303222656, id: 5250), Label: 4960, Image: 49


Query Number: 1120
hit: (distance: 53.442623138427734, id: 2077), Label: 37, Image: 2
hit: (distance: 58.124168395996094, id: 6797), Label: 37, Image: 3
hit: (distance: 156.81939697265625, id: 94), Label: 640, Image: 2
hit: (distance: 157.9536590576172, id: 5593), Label: 4998, Image: 1
hit: (distance: 166.5062713623047, id: 1078), Label: 640, Image: 3
hit: (distance: 168.24169921875, id: 7342), Label: 2831, Image: 1
hit: (distance: 172.74053955078125, id: 6322), Label: 4830, Image: 3
Query Number: 1121
hit: (distance: 74.85426330566406, id: 10165), Label: 5458, Image: 100
hit: (distance: 75.3232421875, id: 7453), Label: 5458, Image: 97
hit: (distance: 76.06666564941406, id: 4615), Label: 5458, Image: 98
hit: (distance: 77.86764526367188, id: 612), Label: 5458, Image: 121
hit: (distance: 78.86952209472656, id: 135), Label: 5458, Image: 95
hit: (distance: 79.6416015625, id: 6041), Label: 5458, Image: 136
hit: (distance: 81.28521728515625, id: 5047), Label: 5458, Image: 141
Query Number: 

hit: (distance: 29.527019500732422, id: 525), Label: 4775, Image: 13
hit: (distance: 32.04274368286133, id: 10162), Label: 4775, Image: 28
hit: (distance: 33.386104583740234, id: 2755), Label: 4775, Image: 17
hit: (distance: 33.38895034790039, id: 4371), Label: 4775, Image: 7
hit: (distance: 33.51734161376953, id: 7794), Label: 4775, Image: 32
hit: (distance: 35.713138580322266, id: 2951), Label: 4775, Image: 8
Query Number: 1262
hit: (distance: 34.707305908203125, id: 6641), Label: 1047, Image: 82
hit: (distance: 36.87094497680664, id: 9542), Label: 1047, Image: 10
hit: (distance: 37.700950622558594, id: 6901), Label: 1047, Image: 73
hit: (distance: 38.138877868652344, id: 3527), Label: 1047, Image: 143
hit: (distance: 38.54020690917969, id: 68), Label: 1047, Image: 58
hit: (distance: 39.348716735839844, id: 5157), Label: 1047, Image: 92
hit: (distance: 39.752220153808594, id: 9765), Label: 1047, Image: 97
Query Number: 1263
hit: (distance: 140.45404052734375, id: 10195), Label: 2365,

Query Number: 1281
hit: (distance: 165.65939331054688, id: 3309), Label: 2979, Image: 2
hit: (distance: 169.10195922851562, id: 1673), Label: 1712, Image: 1
hit: (distance: 173.962646484375, id: 8945), Label: 5251, Image: 1
hit: (distance: 179.36355590820312, id: 2713), Label: 2928, Image: 1
hit: (distance: 179.71067810058594, id: 9836), Label: 1617, Image: 4
hit: (distance: 183.13339233398438, id: 10501), Label: 1364, Image: 1
hit: (distance: 187.04840087890625, id: 8144), Label: 5113, Image: 1
Query Number: 1282
hit: (distance: 95.05357360839844, id: 4219), Label: 2504, Image: 12
hit: (distance: 103.53413391113281, id: 1764), Label: 2504, Image: 2
hit: (distance: 105.85992431640625, id: 1042), Label: 2504, Image: 8
hit: (distance: 106.29010772705078, id: 7074), Label: 2504, Image: 10
hit: (distance: 112.69458770751953, id: 5989), Label: 2504, Image: 7
hit: (distance: 125.72686767578125, id: 3111), Label: 2504, Image: 5
hit: (distance: 128.4853057861328, id: 5895), Label: 2504, Image:

hit: (distance: 95.65585327148438, id: 1957), Label: 999, Image: 1
hit: (distance: 106.53399658203125, id: 3120), Label: 4439, Image: 3
hit: (distance: 107.46688079833984, id: 9759), Label: 1239, Image: 1
Query Number: 1410
hit: (distance: 32.0181770324707, id: 3670), Label: 4532, Image: 1
hit: (distance: 145.62545776367188, id: 8358), Label: 3738, Image: 1
hit: (distance: 151.37203979492188, id: 129), Label: 3737, Image: 1
hit: (distance: 154.79977416992188, id: 3290), Label: 2555, Image: 3
hit: (distance: 157.03050231933594, id: 6773), Label: 3737, Image: 17
hit: (distance: 166.07469177246094, id: 9091), Label: 3737, Image: 8
hit: (distance: 167.8563690185547, id: 2702), Label: 5570, Image: 11
Query Number: 1411
hit: (distance: 68.00621032714844, id: 8288), Label: 4698, Image: 3
hit: (distance: 76.64140319824219, id: 3494), Label: 4698, Image: 5
hit: (distance: 78.28225708007812, id: 5896), Label: 4698, Image: 2
hit: (distance: 82.43049621582031, id: 9256), Label: 4698, Image: 1
hit:

hit: (distance: 154.68338012695312, id: 7333), Label: 3386, Image: 1
Query Number: 1438
hit: (distance: 144.98626708984375, id: 4408), Label: 1929, Image: 1
hit: (distance: 156.4290771484375, id: 2760), Label: 1264, Image: 1
hit: (distance: 163.12246704101562, id: 6673), Label: 3793, Image: 1
hit: (distance: 165.15382385253906, id: 7260), Label: 1393, Image: 103
hit: (distance: 165.54722595214844, id: 850), Label: 370, Image: 8
hit: (distance: 165.79925537109375, id: 1222), Label: 370, Image: 77
hit: (distance: 166.17315673828125, id: 7100), Label: 3175, Image: 5
Query Number: 1439
hit: (distance: 144.10354614257812, id: 10228), Label: 2417, Image: 1
hit: (distance: 147.09912109375, id: 7839), Label: 3152, Image: 1
hit: (distance: 154.69219970703125, id: 5099), Label: 650, Image: 1
hit: (distance: 163.61216735839844, id: 1809), Label: 982, Image: 1
hit: (distance: 175.1580810546875, id: 3609), Label: 109, Image: 1
hit: (distance: 175.82679748535156, id: 5771), Label: 4546, Image: 1
hit

Query Number: 1580
hit: (distance: 115.586669921875, id: 7115), Label: 1891, Image: 69
hit: (distance: 122.25961303710938, id: 3023), Label: 1891, Image: 28
hit: (distance: 129.55020141601562, id: 2636), Label: 1891, Image: 59
hit: (distance: 136.15457153320312, id: 8841), Label: 1891, Image: 38
hit: (distance: 139.8551788330078, id: 9704), Label: 1891, Image: 62
hit: (distance: 140.12863159179688, id: 1671), Label: 1891, Image: 15
hit: (distance: 147.81417846679688, id: 1571), Label: 1891, Image: 103
Query Number: 1581
hit: (distance: 130.3531951904297, id: 1824), Label: 5406, Image: 1
hit: (distance: 145.58352661132812, id: 4985), Label: 484, Image: 1
hit: (distance: 170.52102661132812, id: 2130), Label: 458, Image: 1
hit: (distance: 177.40139770507812, id: 3857), Label: 1574, Image: 1
hit: (distance: 177.53347778320312, id: 8563), Label: 1934, Image: 1
hit: (distance: 179.50714111328125, id: 4222), Label: 4492, Image: 4
hit: (distance: 181.5668487548828, id: 4766), Label: 1602, Imag

hit: (distance: 73.89476776123047, id: 8754), Label: 4551, Image: 29
Query Number: 1601
hit: (distance: 145.6051025390625, id: 4046), Label: 3761, Image: 1
hit: (distance: 150.5596160888672, id: 2462), Label: 3858, Image: 2
hit: (distance: 160.7738037109375, id: 7847), Label: 3239, Image: 1
hit: (distance: 161.21533203125, id: 439), Label: 4519, Image: 4
hit: (distance: 165.41900634765625, id: 145), Label: 5541, Image: 8
hit: (distance: 168.6358184814453, id: 10320), Label: 4519, Image: 2
hit: (distance: 169.95932006835938, id: 9238), Label: 4390, Image: 1
Query Number: 1602
hit: (distance: 100.15206909179688, id: 1895), Label: 3278, Image: 2
hit: (distance: 142.18594360351562, id: 4103), Label: 4297, Image: 1
hit: (distance: 145.57762145996094, id: 2861), Label: 4836, Image: 25
hit: (distance: 149.3708038330078, id: 4887), Label: 1129, Image: 1
hit: (distance: 160.92721557617188, id: 7532), Label: 5407, Image: 1
hit: (distance: 161.54791259765625, id: 8441), Label: 570, Image: 1
hit: 

hit: (distance: 119.07935333251953, id: 3876), Label: 3201, Image: 37
hit: (distance: 123.93622589111328, id: 7380), Label: 3201, Image: 23
Query Number: 1740
hit: (distance: 153.11257934570312, id: 1278), Label: 816, Image: 5
hit: (distance: 161.24380493164062, id: 1276), Label: 3033, Image: 1
hit: (distance: 161.67579650878906, id: 7744), Label: 816, Image: 4
hit: (distance: 166.9815673828125, id: 138), Label: 1012, Image: 1
hit: (distance: 174.68963623046875, id: 1148), Label: 2956, Image: 2
hit: (distance: 175.6383056640625, id: 343), Label: 2502, Image: 3
hit: (distance: 180.75930786132812, id: 1827), Label: 4949, Image: 1
Query Number: 1741
hit: (distance: 171.63479614257812, id: 3654), Label: 5090, Image: 2
hit: (distance: 177.73928833007812, id: 1318), Label: 5001, Image: 1
hit: (distance: 178.9879150390625, id: 9336), Label: 4861, Image: 2
hit: (distance: 183.8582763671875, id: 7442), Label: 1069, Image: 1
hit: (distance: 184.16806030273438, id: 4547), Label: 73, Image: 2
hit:

hit: (distance: 156.0093231201172, id: 2749), Label: 501, Image: 4
hit: (distance: 156.33514404296875, id: 10077), Label: 1047, Image: 134
hit: (distance: 157.40936279296875, id: 934), Label: 3379, Image: 1
hit: (distance: 160.4962158203125, id: 9317), Label: 1047, Image: 133
hit: (distance: 163.54367065429688, id: 5157), Label: 1047, Image: 92
hit: (distance: 164.47547912597656, id: 3413), Label: 1047, Image: 39
hit: (distance: 166.61865234375, id: 6641), Label: 1047, Image: 82
Query Number: 1768
hit: (distance: 90.44550323486328, id: 660), Label: 4867, Image: 2
hit: (distance: 157.23983764648438, id: 8193), Label: 3945, Image: 8
hit: (distance: 162.42935180664062, id: 8563), Label: 1934, Image: 1
hit: (distance: 164.66555786132812, id: 8340), Label: 2444, Image: 2
hit: (distance: 164.91415405273438, id: 9129), Label: 1221, Image: 1
hit: (distance: 166.49867248535156, id: 5501), Label: 2444, Image: 1
hit: (distance: 166.80636596679688, id: 7221), Label: 3945, Image: 3
Query Number: 17

hit: (distance: 33.28045654296875, id: 6549), Label: 5704, Image: 9
hit: (distance: 42.45748519897461, id: 884), Label: 5704, Image: 11
hit: (distance: 49.35321044921875, id: 5761), Label: 5704, Image: 12
hit: (distance: 52.705810546875, id: 2019), Label: 5704, Image: 13
hit: (distance: 59.17024230957031, id: 3784), Label: 5704, Image: 8
hit: (distance: 65.89447784423828, id: 1844), Label: 5704, Image: 3
Query Number: 1910
hit: (distance: 168.99952697753906, id: 6096), Label: 297, Image: 5
hit: (distance: 173.85494995117188, id: 10303), Label: 4457, Image: 2
hit: (distance: 185.8875274658203, id: 5290), Label: 297, Image: 2
hit: (distance: 186.57229614257812, id: 619), Label: 3587, Image: 1
hit: (distance: 187.8704833984375, id: 8756), Label: 1059, Image: 10
hit: (distance: 190.6468048095703, id: 271), Label: 2017, Image: 4
hit: (distance: 191.6046142578125, id: 10213), Label: 1327, Image: 1
Query Number: 1911
hit: (distance: 47.235679626464844, id: 806), Label: 4604, Image: 1
hit: (di

Query Number: 1943
hit: (distance: 143.88836669921875, id: 3805), Label: 2945, Image: 2
hit: (distance: 151.7603759765625, id: 5941), Label: 2945, Image: 1
hit: (distance: 165.083740234375, id: 5131), Label: 2654, Image: 1
hit: (distance: 171.3665008544922, id: 3691), Label: 3270, Image: 9
hit: (distance: 174.45965576171875, id: 3123), Label: 5450, Image: 1
hit: (distance: 174.8057861328125, id: 10236), Label: 3270, Image: 8
hit: (distance: 176.66932678222656, id: 9293), Label: 3270, Image: 4
Query Number: 1944
hit: (distance: 57.160797119140625, id: 500), Label: 3919, Image: 2
hit: (distance: 66.6619873046875, id: 6790), Label: 3919, Image: 1
hit: (distance: 68.57321166992188, id: 1967), Label: 3919, Image: 4
hit: (distance: 156.8910369873047, id: 1416), Label: 1387, Image: 3
hit: (distance: 162.20147705078125, id: 8580), Label: 4341, Image: 1
hit: (distance: 163.1132354736328, id: 2453), Label: 1387, Image: 2
hit: (distance: 164.23074340820312, id: 1790), Label: 2599, Image: 1
Query 

hit: (distance: 138.9375762939453, id: 1360), Label: 1582, Image: 1
hit: (distance: 147.1141357421875, id: 6054), Label: 1850, Image: 1
hit: (distance: 149.84207153320312, id: 77), Label: 3970, Image: 1
hit: (distance: 153.77120971679688, id: 3678), Label: 1047, Image: 21
hit: (distance: 156.4340362548828, id: 1250), Label: 1047, Image: 35
hit: (distance: 156.8955078125, id: 10297), Label: 1811, Image: 1
hit: (distance: 156.96498107910156, id: 10379), Label: 1850, Image: 2
Query Number: 2069
hit: (distance: 46.80610656738281, id: 3822), Label: 2565, Image: 15
hit: (distance: 48.75395584106445, id: 3195), Label: 2565, Image: 18
hit: (distance: 55.026824951171875, id: 6248), Label: 2565, Image: 20
hit: (distance: 69.42767333984375, id: 6398), Label: 2565, Image: 16
hit: (distance: 74.99981689453125, id: 6428), Label: 2565, Image: 7
hit: (distance: 76.57254028320312, id: 8664), Label: 2565, Image: 6
hit: (distance: 80.370849609375, id: 4012), Label: 2565, Image: 3
Query Number: 2070
hit: 

hit: (distance: 56.33833694458008, id: 10334), Label: 2173, Image: 15
hit: (distance: 57.569801330566406, id: 3860), Label: 2173, Image: 11
Query Number: 2097
hit: (distance: 165.23060607910156, id: 1707), Label: 776, Image: 2
hit: (distance: 170.75558471679688, id: 9699), Label: 4055, Image: 1
hit: (distance: 172.36856079101562, id: 8008), Label: 1800, Image: 2
hit: (distance: 173.35987854003906, id: 4423), Label: 2563, Image: 1
hit: (distance: 179.75958251953125, id: 8143), Label: 182, Image: 1
hit: (distance: 180.39303588867188, id: 10509), Label: 3877, Image: 1
hit: (distance: 180.73651123046875, id: 7832), Label: 407, Image: 1
Query Number: 2098
hit: (distance: 144.61656188964844, id: 252), Label: 3571, Image: 3
hit: (distance: 144.6859588623047, id: 7319), Label: 3450, Image: 1
hit: (distance: 146.57513427734375, id: 9713), Label: 708, Image: 1
hit: (distance: 152.0860595703125, id: 8464), Label: 5623, Image: 1
hit: (distance: 152.64950561523438, id: 9870), Label: 5160, Image: 1


hit: (distance: 53.76569366455078, id: 39), Label: 5458, Image: 101
hit: (distance: 55.82567596435547, id: 9118), Label: 5458, Image: 107
hit: (distance: 62.71601104736328, id: 2273), Label: 5458, Image: 117
hit: (distance: 64.4134750366211, id: 697), Label: 5458, Image: 55
hit: (distance: 70.15971374511719, id: 8942), Label: 5458, Image: 64
hit: (distance: 70.99058532714844, id: 6538), Label: 5458, Image: 22
Query Number: 2239
hit: (distance: 151.85818481445312, id: 2940), Label: 2821, Image: 2
hit: (distance: 156.79226684570312, id: 5034), Label: 194, Image: 1
hit: (distance: 157.78453063964844, id: 1077), Label: 2051, Image: 1
hit: (distance: 160.34701538085938, id: 10329), Label: 3875, Image: 1
hit: (distance: 160.97433471679688, id: 1251), Label: 4336, Image: 1
hit: (distance: 168.19183349609375, id: 5415), Label: 2191, Image: 1
hit: (distance: 169.5654296875, id: 8260), Label: 5091, Image: 2
Query Number: 2240
hit: (distance: 64.13137817382812, id: 8352), Label: 5600, Image: 2
hi

hit: (distance: 24.878719329833984, id: 6746), Label: 5250, Image: 2
hit: (distance: 38.103424072265625, id: 10094), Label: 5250, Image: 5
hit: (distance: 42.74183654785156, id: 6090), Label: 5250, Image: 3
hit: (distance: 42.87854766845703, id: 2050), Label: 5250, Image: 8
hit: (distance: 46.649635314941406, id: 2890), Label: 5250, Image: 6
hit: (distance: 49.992950439453125, id: 3672), Label: 5250, Image: 10
hit: (distance: 50.5277214050293, id: 1953), Label: 5250, Image: 11
Query Number: 2271
hit: (distance: 148.23748779296875, id: 5065), Label: 4110, Image: 1
hit: (distance: 148.91542053222656, id: 1468), Label: 4335, Image: 1
hit: (distance: 150.93972778320312, id: 3616), Label: 4109, Image: 1
hit: (distance: 151.89212036132812, id: 9488), Label: 926, Image: 1
hit: (distance: 152.89291381835938, id: 1171), Label: 3900, Image: 2
hit: (distance: 156.52658081054688, id: 752), Label: 2573, Image: 1
hit: (distance: 157.52584838867188, id: 583), Label: 3900, Image: 1
Query Number: 2272


hit: (distance: 119.24394989013672, id: 4420), Label: 4686, Image: 6
hit: (distance: 147.2813720703125, id: 3374), Label: 4836, Image: 19
Query Number: 2429
hit: (distance: 140.57350158691406, id: 2868), Label: 189, Image: 6
hit: (distance: 144.88746643066406, id: 3823), Label: 2182, Image: 1
hit: (distance: 146.2556610107422, id: 3864), Label: 4865, Image: 1
hit: (distance: 146.52056884765625, id: 4636), Label: 4974, Image: 1
hit: (distance: 152.66009521484375, id: 8615), Label: 784, Image: 2
hit: (distance: 154.24276733398438, id: 615), Label: 189, Image: 1
hit: (distance: 158.81069946289062, id: 4089), Label: 17, Image: 4
Query Number: 2430
hit: (distance: 147.1808624267578, id: 1424), Label: 2895, Image: 1
hit: (distance: 153.26461791992188, id: 3013), Label: 4770, Image: 1
hit: (distance: 165.1093292236328, id: 5705), Label: 1992, Image: 2
hit: (distance: 166.0082244873047, id: 8555), Label: 4615, Image: 1
hit: (distance: 168.76205444335938, id: 5663), Label: 3462, Image: 2
hit: (

Query Number: 2565
hit: (distance: 107.87322998046875, id: 2160), Label: 813, Image: 1
hit: (distance: 144.3643035888672, id: 6027), Label: 1059, Image: 4
hit: (distance: 153.66317749023438, id: 7829), Label: 1059, Image: 5
hit: (distance: 155.93553161621094, id: 9161), Label: 368, Image: 1
hit: (distance: 158.04290771484375, id: 7208), Label: 1059, Image: 2
hit: (distance: 159.048095703125, id: 643), Label: 469, Image: 1
hit: (distance: 161.23162841796875, id: 9136), Label: 1059, Image: 6
Query Number: 2566
hit: (distance: 47.93610382080078, id: 3698), Label: 3270, Image: 5
hit: (distance: 52.0937614440918, id: 10236), Label: 3270, Image: 8
hit: (distance: 54.315853118896484, id: 9293), Label: 3270, Image: 4
hit: (distance: 74.51268005371094, id: 8530), Label: 3270, Image: 1
hit: (distance: 77.12777709960938, id: 6071), Label: 3270, Image: 7
hit: (distance: 87.8106689453125, id: 3691), Label: 3270, Image: 9
hit: (distance: 167.8284454345703, id: 1331), Label: 3892, Image: 3
Query Numb

hit: (distance: 40.08051300048828, id: 6575), Label: 3104, Image: 7
hit: (distance: 53.9455680847168, id: 3045), Label: 3104, Image: 6
hit: (distance: 73.26724243164062, id: 9971), Label: 3104, Image: 4
Query Number: 2599
hit: (distance: 67.03400421142578, id: 8699), Label: 1870, Image: 466
hit: (distance: 71.30258178710938, id: 8191), Label: 1870, Image: 355
hit: (distance: 72.0323715209961, id: 9664), Label: 1870, Image: 467
hit: (distance: 72.23062896728516, id: 2629), Label: 1870, Image: 184
hit: (distance: 74.91110229492188, id: 163), Label: 1870, Image: 194
hit: (distance: 75.728515625, id: 2484), Label: 1870, Image: 515
hit: (distance: 75.73306274414062, id: 1415), Label: 1870, Image: 431
Query Number: 2600
hit: (distance: 162.55735778808594, id: 9066), Label: 3559, Image: 1
hit: (distance: 165.92922973632812, id: 3410), Label: 653, Image: 1
hit: (distance: 166.97763061523438, id: 7933), Label: 3238, Image: 2
hit: (distance: 171.34396362304688, id: 7746), Label: 963, Image: 1
hi

In [94]:
final_result

{'173_0002': {'173_0001': 26.057170867919922,
  '999_0001': 127.43841552734375,
  '1105_0002': 130.5911865234375,
  '2368_0001': 130.891357421875,
  '1610_0001': 131.17236328125,
  '3250_0001': 135.7631378173828,
  '2368_0002': 136.7291259765625},
 '4836_0026': {'4836_0008': 37.43260192871094,
  '4836_0013': 53.6215934753418,
  '4836_0019': 59.88230895996094,
  '4836_0005': 60.98465347290039,
  '4836_0009': 63.946720123291016,
  '4836_0023': 66.75064086914062,
  '4836_0017': 67.14411926269531},
 '1428_0002': {'1428_0001': 117.2981948852539,
  '4769_0001': 136.78451538085938,
  '5299_0001': 152.08587646484375,
  '3125_0001': 159.51443481445312,
  '2579_0001': 164.4651336669922,
  '4348_0001': 165.86683654785156,
  '2661_0003': 166.17874145507812},
 '5367_0016': {'5367_0010': 87.85360717773438,
  '5367_0014': 92.42774200439453,
  '5367_0002': 120.11233520507812,
  '5367_0017': 129.45809936523438,
  '5367_0009': 129.7062530517578,
  '5367_0012': 132.98597717285156,
  '5367_0004': 133.9378

In [95]:
import json

# Serialize data into file - to store the output for visualization purposes
json.dump( final_result, open( "result_seven.json", 'w' ) )

In [98]:
# Read data from file:
data = json.load( open( "result_seven.json" ) )

In [99]:
data

{'173_0002': {'173_0001': 26.057170867919922,
  '999_0001': 127.43841552734375,
  '1105_0002': 130.5911865234375,
  '2368_0001': 130.891357421875,
  '1610_0001': 131.17236328125,
  '3250_0001': 135.7631378173828,
  '2368_0002': 136.7291259765625},
 '4836_0026': {'4836_0008': 37.43260192871094,
  '4836_0013': 53.6215934753418,
  '4836_0019': 59.88230895996094,
  '4836_0005': 60.98465347290039,
  '4836_0009': 63.946720123291016,
  '4836_0023': 66.75064086914062,
  '4836_0017': 67.14411926269531},
 '1428_0002': {'1428_0001': 117.2981948852539,
  '4769_0001': 136.78451538085938,
  '5299_0001': 152.08587646484375,
  '3125_0001': 159.51443481445312,
  '2579_0001': 164.4651336669922,
  '4348_0001': 165.86683654785156,
  '2661_0003': 166.17874145507812},
 '5367_0016': {'5367_0010': 87.85360717773438,
  '5367_0014': 92.42774200439453,
  '5367_0002': 120.11233520507812,
  '5367_0017': 129.45809936523438,
  '5367_0009': 129.7062530517578,
  '5367_0012': 132.98597717285156,
  '5367_0004': 133.9378

# Realse Collection

In [101]:
utility.drop_collection("embed") # Not Important but to drop the collection we just created

In [ ]:
# python3 hello_milvus.py

# Accuracy

In [14]:
import json
import os
from sklearn.metrics import accuracy_score

In [4]:
data = json.load( open(os.path.join("0004_milvus", "result_three.json")))

In [15]:
y_test.astype(int)

array([ 173, 4836, 1428, ..., 5354, 3109, 1393])

In [27]:
# load the saved outputs to calculate accuracy
y_pred = list()
for index, result in data.items():
    
    image = list()
    label = list()
    dist = list()
    
    for result_index, result_dist in result.items():
        label.append(result_index.split('_')[0])
        image.append(result_index.split('_')[1])
        dist.append(result_dist)
    y_pred.append(int(max(label,key=label.count)))
    print(image, label, dist, sep='\n')

y_pred = np.array(y_pred)

['0001', '0001', '0002']
['173', '999', '1105']
[26.057170867919922, 127.43841552734375, 130.5911865234375]
['0008', '0013', '0019']
['4836', '4836', '4836']
[37.43260192871094, 53.6215934753418, 59.88230895996094]
['0001', '0001', '0001']
['1428', '4769', '5299']
[117.2981948852539, 136.78451538085938, 152.08587646484375]
['0010', '0014', '0002']
['5367', '5367', '5367']
[87.85360717773438, 92.42774200439453, 120.11233520507812]
['0024', '0002', '0002']
['3401', '3914', '4380']
[148.79164123535156, 155.47116088867188, 156.21974182128906]
['0003', '0002', '0002']
['4631', '4631', '1966']
[154.74160766601562, 155.43798828125, 162.2541046142578]
['0004', '0001', '0002']
['5258', '5258', '5258']
[55.15113067626953, 60.07026290893555, 61.48064422607422]
['0001', '0002', '0002']
['4882', '4882', '3929']
[167.69810485839844, 183.77108764648438, 201.8341522216797]
['0001', '0010', '0002']
['4003', '4001', '4003']
[147.05587768554688, 153.54180908203125, 157.07298278808594]
['0002', '0001', '0

[175.0430908203125, 182.35047912597656, 191.53187561035156]
['0004', '0007', '0005']
['1696', '1696', '1696']
[80.97842407226562, 105.23661804199219, 108.58224487304688]
['0001', '0009', '0007']
['5400', '5400', '5400']
[92.28768157958984, 98.13172149658203, 99.55165100097656]
['0014', '0013', '0024']
['386', '386', '386']
[47.165191650390625, 77.46882629394531, 80.70758819580078]
['0006', '0008', '0003']
['5219', '5219', '5219']
[71.21333312988281, 105.11368560791016, 134.889892578125]
['0011', '0008', '0001']
['4972', '4972', '4972']
[102.13230895996094, 116.90585327148438, 124.04317474365234]
['0004', '0002', '0002']
['4228', '4228', '1508']
[86.03629302978516, 92.91939544677734, 189.09503173828125]
['0005', '0004', '0002']
['4911', '4911', '1558']
[101.09368896484375, 147.58554077148438, 162.4354248046875]
['0001', '0002', '0001']
['3469', '14', '5686']
[128.21644592285156, 155.35308837890625, 161.3409423828125]
['0002', '0001', '0001']
['1476', '4961', '3355']
[136.8139190673828, 

In [28]:
accuracy_score(y_test, y_pred)

0.6218360408009067

### Here you can see that we received an accuracy of 62%. Given that LFW have many people with 2 or less images this is good and much better than our previous approaches.